browser/src/control/jsdialog/Util.KeyboardGridNavigation.ts

```diff
@@ -124,10 +129,27 @@ function moveFocus(
 	}
 }
 
-function getRowColumn(element: HTMLElement): [number, number] {
+function getRowColumn(
+	element: HTMLElement,
+	parentContainer: HTMLElement,
+): [number, number] {
 	const index = element.getAttribute('index');
-	if (!index) return [-1, -1]; // we will never have this kind of index this is why we are pssing nagative values here
-	const [row, column] = index.split(':').map(Number);
+	if (!index) return [-1, -1]; // we will never have this kind of index this is why we are passing negative values here
+
+	if (index.includes(':')) {
+		const [row, column] = index.split(':').map(Number);
+		return [row, column];
+	}
+
+	const flatIndex = index.trim() !== '' ? Number(index.trim()) : NaN;
+	const gridTemplateColumns =
+		getComputedStyle(parentContainer).gridTemplateColumns;
```

Comment:
This is very expensive operation we should avoid if possible.
Which case does it handle? Can't we somehow avoid it and use previous method too?